src/AutoRest.CSharp/Mgmt/Output/MgmtObjectType.cs

```diff
@@ -93,6 +93,8 @@ private ObjectTypeProperty CreatePropertyType(ObjectTypeProperty objectTypePrope
         protected override CSharpType? CreateInheritedType()
         {
             CSharpType? inheritedType = base.CreateInheritedType();
+            if (inheritedType?.IsFrameworkType == true && !inheritedType.FrameworkType.IsGenericType)
+                return inheritedType;
 
             var typeToReplace = inheritedType?.Implementation as MgmtObjectType;
```

Comment:
When testing ResourceLinks and `inheritedType` is `SubResource`, it is a FrameworkType and calling `inheritedType?.Implementation` will throw an exception: `Not implemented type: Azure.ResourceManager.Resources.Models.SubResource`. So adding a check here, if inheritedType is already a non-generic-type framewrok type (not List/Dictionary), we can return it directly.